src/DateInput.tsx

```diff
@@ -358,3 +360,84 @@ export function DateInputFieldWithLabel (
     </>
   )
 }
+
+// Type-in Date Input
+
+const dateMask = [/\d/, /\d/, '/', /\d/, /\d/, '/', /\d/, /\d/, /\d/, /\d/]
+
+export function DateInputTypeIn ({
+  name,
+  hasError,
+  onChange,
+  onBlur
+}: {
+  name: string
+  hasError?: boolean
+  onChange(val: string): void
+  onBlur(): void
+}) {
+  return (
+    <MaskedInput
+      name={name}
+      mask={dateMask}
+      id='date'
+      placeholder='mm/dd/yyyy'
+      type='text'
+      onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
+        onChange(e.target.value)
+      }}
+      onBlur={onBlur}
+      render={(inputRef, props) => (
+        <Input
+          name={name}
+          ref={inputRef}
+          hasError={hasError}
+          onChange={onChange}
+          onBlur={onBlur}
+          {...props}
+        />
+      )}
+    />
+  )
+}
+
+export function DateInputTypeInField ({ name }: { name: string }) {
+  const validate = (value: string) => {
```

Comment:
Let's wrap this in a useCallback

Reply:
Gotcha, I want to understand the benefits of this: is the main benefit of using `useCallback` to prevent us from generating this function on each rerender?

Reply:
Yep, just can prevent re-renders in some cases. Not a huge benefit probably in this case, but good hygiene.